src/constitutive/TACSBladeStiffenedShellConstitutive.cpp

```diff
@@ -2646,15 +2705,166 @@ TacsScalar TACSBladeStiffenedShellConstitutive::computeCriticalShearLoad(
+  for (int ii = 0; ii < 4; ii++) {
+    localBucklingDMatSens[ii] =
+        N1CritSens[ii] * dfdN1CritLocal + N12CritSens[ii] * dfdN12CritLocal;
+  }
+
+  // --- Panel thickness sensitivity ---
+  if (this->panelThickNum >= 0) {
+    const int dvNum = this->panelThickLocalNum;
+    // --- Local buckling contribution ---
+    const TacsScalar dMatdt = 0.25 * t * t;  // d/dt(t^3/12) = t^2/4
+    for (int ii = 0; ii < this->numPanelPlies; ii++) {
+      TacsScalar* Q = &(this->panelQMats[ii * NUM_Q_ENTRIES]);
+      dfdx[dvNum] +=
+          scale * dMatdt * this->panelPlyFracs[ii] *
+          (localBucklingDMatSens[0] * Q[0] + localBucklingDMatSens[1] * Q[3] +
+           localBucklingDMatSens[2] * Q[1] + localBucklingDMatSens[3] * Q[5]);
+    }
+  }
+
+  // --- Panel Ply fraction sensitivities ---
+  for (int ii = 0; ii < this->numPanelPlies; ii++) {
+    if (this->panelPlyFracNums[ii] >= 0) {
+      const int dvNum = this->panelPlyFracLocalNums[ii];
+      // --- Local buckling contribution ---
+      const TacsScalar DFactor = t * t * t / 12.0;
+      const TacsScalar* const Q = &(this->panelQMats[ii * NUM_Q_ENTRIES]);
+      // Do df/dx += df/dD11 * dD11/dx + df/dD22 * dD22/dx + df/dD12 * dD12/dx
+      // + df/dD66 * dD66/dx
+      dfdx[dvNum] +=
+          scale * DFactor *
+          (localBucklingDMatSens[0] * Q[0] + localBucklingDMatSens[1] * Q[3] +
+           localBucklingDMatSens[2] * Q[1] + localBucklingDMatSens[3] * Q[5]);
+    }
+  }
+
+  // --- Stiffener pitch sensitivity ---
+  if (this->stiffenerPitchNum >= 0) {
+    dfdx[this->stiffenerPitchLocalNum] +=
+        scale *
+        (N12CritSens[4] * dfdN12CritLocal + N1CritSens[4] * dfdN1CritLocal);
+  }
+}
+
+TacsScalar
+TACSBladeStiffenedShellConstitutive::computeCriticalLocalAxialLoadSens(
+    const TacsScalar D11, const TacsScalar D22, const TacsScalar D12,
+    const TacsScalar D66, const TacsScalar L, TacsScalar* const D11Sens,
+    TacsScalar* const D22Sens, TacsScalar* const D12Sens,
+    TacsScalar* const D66Sens, TacsScalar* const LSens) {
+  const double pi2 = M_PI * M_PI;
+  const TacsScalar L2 = L * L;
+  const TacsScalar root = sqrt(D11 * D22);
+
+  *D11Sens = pi2 * root / (D11 * L2);
+  *D12Sens = 2.0 * pi2 / L2;
+  *D22Sens = pi2 * root / (D22 * L2);
+  *D66Sens = 4.0 * pi2 / L2;
+  *LSens = 4.0 * pi2 * (-D12 - 2.0 * D66 - root) / (L2 * L);
+
+  return 2.0 * pi2 / L2 * (root + D12 + 2.0 * D66);
+}
+
 TacsScalar TACSBladeStiffenedShellConstitutive::computeCriticalShearLoadSens(
     const TacsScalar D1, const TacsScalar D2, const TacsScalar D3,
-    const TacsScalar L, TacsScalar* D1Sens, TacsScalar* D2Sens,
-    TacsScalar* D3Sens, TacsScalar* LSens) {
-  TacsScalar L2 = L * L;
-  TacsScalar L3 = L2 * L;
-  TacsScalar D32 = D3 * D3;
-  TacsScalar root = sqrt(D1 * D2);
-  TacsScalar xi = root / D3;
+    const TacsScalar L, TacsScalar* const D1Sens, TacsScalar* const D2Sens,
+    TacsScalar* const D3Sens, TacsScalar* const LSens) {
```

Comment:
This is not that important as long as the test works though.

Reply:
I think these equations are going to be pretty gnarly however they're written, even the power series method is not so straightforward because all of the equations depend on both D1/2/3 and xi which is a function of D1/2/3 so it's not obvious what the exponent is.